lib/temporalio/worker/activity_worker.rb

```diff
@@ -37,8 +38,19 @@ def run(reactor)
       rescue Temporalio::Bridge::Error::WorkerShutdown
         # No need to re-raise this error, it's a part of a normal shutdown
       ensure
-        reactor.async do
+        reactor.async do |async_task|
+          cancelation_task =
+            if graceful_timeout
+              async_task.async do
+                sleep graceful_timeout
+                running_activities.each_value do |activity_runner|
+                  activity_runner.cancel('Worker is shutting down', by_request: false)
+                end
+              end
+            end
```

Comment:
Shouldn't this be done on WorkerShutdown error instead of in this finally block?

Reply:
Hmm… well we do want an orderly shutdown even in the case of a fatal error, right?

Reply:
Probably not, on fatal error, you can't really determine what the state of Core is so it might not even be able to process activity completions.